File: Java/프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        StringBuilder answer = new StringBuilder();

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);

            if ('A' <= c && c <= 'Z') {
                c = (char)((c - 'A' + n) % 26 + 'A');
            } else if ('a' <= c && c <= 'z') {
                c = (char)((c - 'a' + n) % 26 + 'a');
            }
            answer.append(c);
        }
        return answer.toString();
    }
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.kt
class Solution {
    fun solution(s: String, n: Int): String {
        var answer = ""
        var arr = s.toCharArray()
        for (i in arr.indices) {
            arr[i] = when {
                arr[i] in 'a'..'z' -> ((arr[i] - 'a' + n) % 26 + 'a'.toInt()).toChar()
                arr[i] in 'A'..'Z' -> ((arr[i] - 'A' + n) % 26 + 'A'.toInt()).toChar()
                else -> arr[i]
            }
        }
        answer = arr.joinToString("")
        return answer
    }
}